src/test/java/bio/terra/service/filedata/azure/util/BlobIOTestUtility.java

```diff
@@ -0,0 +1,157 @@
+    this.destinationBlobContainerClient.create();
+    this.tokenCredential = tokenCredential;
+  }
+
+  private BlobContainerClient createBlobContainerClient(
+      TokenCredential credential, String containerName, String accountName) {
+
+    return new BlobContainerClientBuilder()
+        .credential(credential)
+        .endpoint(String.format(Locale.ROOT, STORAGE_ENDPOINT_PATTERN, accountName, containerName))
+        .buildClient();
+  }
+
+  public String generateSourceContainerUrlWithSasReadAndListPermissions(String accountKey) {
+    String sasToken = generateContainerSasTokenWithReadAndListPermissions(accountKey);
+
+    return String.format(
+        "%s?%s", this.getSourceBlobContainerClient().getBlobContainerUrl(), sasToken);
+  }
+
+  public String generateContainerSasTokenWithReadAndListPermissions(String accountKey) {
+    BlobSasPermission permissions =
+        new BlobSasPermission().setReadPermission(true).setListPermission(true);
+
+    OffsetDateTime expiryTime = OffsetDateTime.now().plusDays(1);
+    SasProtocol sasProtocol = SasProtocol.HTTPS_ONLY;
+
+    // build the token
+    BlobServiceSasSignatureValues sasSignatureValues =
+        new BlobServiceSasSignatureValues(expiryTime, permissions).setProtocol(sasProtocol);
+
+    BlobContainerClient blobContainerClient =
+        new BlobContainerClientBuilder()
+            .credential(
+                new StorageSharedKeyCredential(
+                    this.sourceBlobContainerClient.getAccountName(), accountKey))
+            .endpoint(this.sourceBlobContainerClient.getBlobContainerUrl())
+            .buildClient();
+
+    return blobContainerClient.generateSas(sasSignatureValues);
+  }
+
+  private String generateNewTempContainerName() {
+    return UUID.randomUUID().toString().toLowerCase().replace("-", "").substring(0, 30);
+  }
+
+  public List<String> uploadSourceFiles(int numOfFiles, long length) {
+    return Stream.iterate(0, n -> n + 1)
+        .limit(numOfFiles)
+        .map(
+            i ->
+                uploadSourceFile(
+                    String.format("%s/%s%s", i, SOURCE_BLOB_NAME, UUID.randomUUID().toString()),
+                    length))
+        .collect(Collectors.toList());
+  }
+
+  public String uploadSourceFile(String blobName, long length) {
+    sourceBlobContainerClient.getBlobClient(blobName).upload(createInputStream(length), length);
+    return blobName;
+  }
+
+  public String uploadDestinationFile(String blobName, long length) {
+    destinationBlobContainerClient
+        .getBlobClient(blobName)
+        .upload(createInputStream(length), length);
+    return blobName;
+  }
+
+  public String getSourceContainerEndpoint() {
+    return this.sourceBlobContainerClient.getBlobContainerUrl();
+  }
+
+  public String getDestinationContainerEndpoint() {
+    return this.destinationBlobContainerClient.getBlobContainerUrl();
+  }
+
+  private InputStream createInputStream(long length) {
+    return new InputStream() {
```

Comment:
You should repace this with `RandomStringUtils`

Reply:
Not sure of the benefit. Please note that this approach minimizes allocations. You have the allocation of the buffer internally created by the HTTP request (the content body). There is little you can do about this. However, the implementation creates a stream without an additional buffer (the whole string pre-allocated). Hence a bit better. If the suggestion is to use RandomStringUtils to create String of X size, then that means that we are allocating such buffer, create a stream out of that, and then have the internal HTTP body be recreated out of that stream. Not sure what that would buy us. Happy to change it if I am missing something though...